docs/http/http-spans.md

```diff
@@ -295,52 +298,56 @@ See the examples for more details about:
+HTTP server frameworks and their instrumentations have a limited knowledge about the HTTP infrastructure and intermediaries request went through. In a general case, they can only use HTTP request properties such as request target or headers to populate `server.*` attributes.
 
-An application can be "mounted" under an **application root**
-(also known as a *[context root][]*, *[context prefix][]*, or *[document base][]*)
-which is a fixed path prefix of the URL that determines to which application a request is routed
-(e.g., the server could be configured to route all requests that go to an URL path starting with `/webshop/`
-at a particular virtual host
-to the `com.example.webshop` web application).
+Certain web frameworks may provide additional information about server. For example CGI specification defines `SERVER_NAME` and `SERVER_PORT` ([RFC 3875][rfc-servername]) variables. HTTP servers usually know server names as they are provided by users in the configuration. See for example, the Apache [`ServerName`][ap-sn] or NGINX [`server_name`][nx-sn] directive.
 
-Some servers allow to bind the same HTTP application to multiple `(virtual host, application root)` pairs.
+Still, the original logical service name might not be available to the server HTTP instrumentation and instrumentations MAY provide an option to let users set `server.*` attributes.
 
-> TODO: Find way to trace HTTP application and application root ([opentelemetry/opentelementry-specification#335][])
+#### Setting `server.address` and `server.port` attributes
 
-[HTTP host header]: https://tools.ietf.org/html/rfc7230#section-5.4
-[PEP 3333]: https://www.python.org/dev/peps/pep-3333/
-[modwsgisetup]: https://modwsgi.readthedocs.io/en/develop/user-guides/quick-configuration-guide.html
-[context root]: https://docs.jboss.org/jbossas/guides/webguide/r2/en/html/ch06.html
-[context prefix]: https://marc.info/?l=apache-cvs&m=130928191414740
-[document base]: http://tomcat.apache.org/tomcat-5.5-doc/config/context.html
+HTTP server instrumentations SHOULD do the best effort when populating `server.address` and `server.port` attributes and determine them by using the first of the following that applies:
+
+* Explicit user configuration (if supported).
+* Server name and port available through web framework or HTTP server configuration (if known to the instrumentation).
+* The [request target](https://www.rfc-editor.org/rfc/rfc9110.html#target.resource) if it's passed in the absolute form.
+* The original host which may be passed by the reverse proxy in the `host` identifier of the [`Forwarded`][Forwarded] header or in the [`X-Forwarded-Host`][X-Forwarded-Host] header.
+* The `Host` header.
+
+As a result, `server.*` attributes capture logical host and port name and intended to be the same on the client and server sides. Application developers MAY capture private host information using applicable [resource attributes](/docs/resource/README.md).
```

Comment:
this might be nice up at the top instead, as it provides the motivation for our definition